Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -251,16 +248,17 @@ void CDMInstanceOpenCDM::requestLicense(LicenseType licenseType, const AtomicStr
     }
 
     String sessionIdValue = String::fromUTF8(sessionId.c_str());
-    std::pair<std::map<std::string, Session>::iterator, bool> newEntry =
-        m_sessionsMap.emplace(std::piecewise_construct, std::forward_as_tuple(sessionId), std::forward_as_tuple(openCDM, WTFMove(initData)));
-
-    const Session& newSession(newEntry.first->second);
+    Ref<Session> newSession = Session::create(openCDM, WTFMove(initData));
+    if (!addSession(sessionIdValue, newSession.ptr())) {
+        GST_WARNING("Failed to add session %s, the session might already exist, or the allocation failed", sessionId.c_str());
+        return;
+    }
 
-    if (newSession.isValid()) {
+    if (newSession->isValid()) {
```

Comment:
I wonder if we should check for the session to be valid or invalid before we add it to the hashmap, shouldn't we? Doing it like now made sense when the objects were created inside the map, but I don't think that's interesting anymore and it could leave to unlikely cases were we add invalid sessions to the map.